converter/ena/ena_run.py

```diff
@@ -0,0 +1,71 @@
+import logging
+from config import ENA_FTP_DIR
+from converter.ena.classes import Run, RunType, FileFiletype, RunSet, AttributeType
+from converter.ena.classes.sra_common import IdentifierType, PlatformType, RefObjectType, TypeIlluminaModel
+from converter.ena.classes.sra_experiment import Experiment, ExperimentType, LibraryDescriptorType, LibraryType, SampleDescriptorType, TypeLibrarySelection, TypeLibrarySource, TypeLibraryStrategy
+
+
+class EnaRun:
+
+    FILE_CHECKSUM_METHOD = 'MD5'
+    SEQUENCE_FILE_TYPES = ['fq', 'fastq', 'fq.gz', 'fastq.gz']
+
+    def __init__(self, hca_data):
+        self.logger = logging.getLogger(__name__)
+        self.submission = hca_data.submission
+        self.assays = hca_data.submission["assays"]
+
+        self.ena_upload_area_path = f'{ENA_FTP_DIR}/{self.submission["uuid"]["uuid"]}/'
+
+    def run_set(self):
+        run_set = RunSet()
+        for assay in self.assays:
+            run_set.run.append(self.run(assay))
+
+        return run_set
+
+    def run(self, assay):
+
+        sequencing_protocol = assay["sequencing_protocol"]
+        #library_preparation_protocol = assay["library_preparation_protocol"]
+        derived_files = assay["derived_files"]
+
+        run = Run()
+        run.run_attributes = Run.RunAttributes()
+        run.run_attributes.run_attribute.append(AttributeType(tag="Description", value=sequencing_protocol["content"]["protocol_core"]["protocol_description"]))
+
+        run.title = assay["content"]["process_core"]["process_id"]
+        run.experiment_ref = RefObjectType()
+        run.experiment_ref.accession = ''
+        run.alias = assay["uuid"]["uuid"]
+
+        run.data_block = RunType.DataBlock()
+        run.data_block.files = RunType.DataBlock.Files()
```

Comment:
Line 33-43 only deals with setting up the `run` object. You could extract that to a private method.

Reply:
thanks @ke4 surely this can be simplified. I'll keep it this way for now as creating the run object involves extracting and combining data from a few sources. I'll look for these reorg when the submit part if working, as this part may still change.